lib/Service/ModelService.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * @copyright Copyright (c) 2019, Matias De lellis <mati86dl@gmail.com>
+ *
+ * @author Matias De lellis <mati86dl@gmail.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\FaceRecognition\Service;
+
+use OCP\IConfig;
+use OCP\Files\IAppData;
+use OCP\Files\IRootFolder;
+
+class ModelService {
+
+	/** @var IConfig */
+	private $config;
+
+	/** @var IAppData */
+	private $appData;
+
+	/** @var IRootFolder */
+	private $rootFolder;
+
+	/** @var string */
+	private $modelsFolder;
+
+	public function __construct(IConfig     $config,
+	                            IAppData    $appData,
+	                            IRootFolder $rootFolder) {
+		$this->config     = $config;
+		$this->appData    = $appData;
+		$this->rootFolder = $rootFolder;
+
+		$this->prepareAppDataFolders();
+	}
+
+	/**
+	 * @return none
+	 */
+	public function useModelVersion (int $version) {
+		try {
+			$this->appData->getFolder('/models/' . $version);
+		} catch (NotFoundException $e) {
+			$this->appData->newFolder('/models/' . $version);
+		}
+
+		$instanceId = $this->config->getSystemValue('instanceid', null);
+		$appData = $this->rootFolder->get('appdata_'.$instanceId)->getPath();
+		$dataDir = $this->config->getSystemValue('datadirectory', null);
+
+		$this->modelsFolder = $dataDir . $appData . '/facerecognition/models/' . $version . '/';
+	}
+
+	/**
+	 * @return String
+	 */
+	public function getModelPath(string $file): string {
+		return $this->modelsFolder . $file;
+	}
+
+	/**
+	 * @return none
+	 */
+	private function prepareAppDataFolders() {
+		try {
+			$this->appData->getFolder('/models');
+		} catch (NotFoundException $e) {
+			$this->appData->newFolder('/models');
```

Comment:
I don't have this exception caught unless I add `use OCP\Files\NotFoundException;` at the beggining. I have something like this (consider adding `use`)

```
$ ./occ
An unhandled exception has been thrown:
OCP\Files\NotFoundException: /appdata_ocwys0rg5cm7/facerecognition/models in /home/branko/server/lib/private/Files/Node/Root.php:202
Stack trace:
#0 [internal function]: OC\Files\Node\Root->get('/appdata_ocwys0...')
#1 /home/branko/server/lib/private/Files/Node/LazyRoot.php(64): call_user_func_array(Array, Array)
#2 /home/branko/server/lib/private/Files/Node/LazyRoot.php(141): OC\Files\Node\LazyRoot->__call('get', Array)
#3 /home/branko/server/lib/private/Files/AppData/AppData.php(136): OC\Files\Node\LazyRoot->get('appdata_ocwys0r...')
#4 /home/branko/server/apps/facerecognition/lib/Service/ModelService.php(83): OC\Files\AppData\AppData->getFolder('/models')
#5 /home/branko/server/apps/facerecognition/lib/Service/ModelService.php(51): OCA\FaceRecognition\Service\ModelService->prepareAppDataFolders()
#6 [internal function]: OCA\FaceRecognition\Service\ModelService->__construct(Object(OC\AllConfig), Object(OC\Files\AppData\AppData), Object(OC\Files\Node\LazyRoot))
#7 /home/branko/server/lib/private/AppFramework/Utility/SimpleContainer.php(81): ReflectionClass->newInstanceArgs(Array)
#8 /home/branko/server/lib/private/AppFramework/Utility/SimpleContainer.php(98): OC\AppFramework\Utility\SimpleContainer->buildClass(Object(ReflectionClass))
#9 /home/branko/server/lib/private/AppFramework/Utility/SimpleContainer.php(119): OC\AppFramework\Utility\SimpleContainer->resolve('OCA\\FaceRecogni...')
#10 /home/branko/server/lib/private/AppFramework/DependencyInjection/DIContainer.php(425): OC\AppFramework\Utility\SimpleContainer->query('OCA\\FaceRecogni...')
#11 /home/branko/server/lib/private/AppFramework/DependencyInjection/DIContainer.php(396): OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback('OCA\\FaceRecogni...')
#12 /home/branko/server/lib/private/AppFramework/Utility/SimpleContainer.php(68): OC\AppFramework\DependencyInjection\DIContainer->query('OCA\\FaceRecogni...')
#13 /home/branko/server/lib/private/AppFramework/Utility/SimpleContainer.php(98): OC\AppFramework\Utility\SimpleContainer->buildClass(Object(ReflectionClass))
```

Reply:
Done..